File: LinearDataStructures/Array/Bubble_Sort.py
def bubble_sort(arr):

    unsorted_until_index = len(arr) - 1
    sorted = False

    while not sorted:
        sorted = True
        
        for i in range(unsorted_until_index):

            if(arr[i]>arr[i+1]):
                temp = arr[i]
                arr[i] = arr[i+1]
                arr[i+1] = temp
                sorted = False
        unsorted_until_index -= 1

arr = [4,3,2,1]
bubble_sort(arr)

for i in arr:
    print(i, end= " ")

File: LinearDataStructures/Array/BubbleSort.java
public class BubbleSort {
    
    public static void bubbleSort(int arr[]){

        int unsortedUntilIndex = arr.length - 1;
        boolean sorted = false;

        while (!sorted) {
            sorted = true;

            for(int i = 0; i<unsortedUntilIndex; i++){

                if (arr[i]>arr[i+1]) {
                    
                    int temp = arr[i];
                    arr[i] = arr[i+1];
                    arr[i+1] = temp;
                    sorted = false;
                }
            }
            unsortedUntilIndex--;
        }

    }

    public static void main(String[] args){

        int arr[] = {4,3,2,1};
        bubbleSort(arr);

        System.out.println("Bubble Sort: ");
        for(int i = 0; i <= arr.length-1; i++){
            System.out.print(arr[i] + " ");
        }

    }
}
